### PR TITLE
Add parameter selection for specialty plots

### PR DESCRIPTION
**Pull request by Bobfrat**
Adds error message to plot and filters the metadata from the streams.
@birdage @DanielJMaher please review

### CODE REVIEW

**Comment by DanielJMaher** [comment]
@birdage Your call on merge.

File: ooiui/static/js/views/common/SVGView.js [hunk 1]
```diff
@@ -112,9 +112,30 @@ var plotParameters ={
   },
   invalidParam:function(){return !this._selectedX.valid},
   validateSelected:function(options){
+    console.log(options.yvar[0][0])
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat remove console log pls

File: ooiui/static/js/views/common/SVGView.js [hunk 2]
```diff
@@ -112,9 +112,30 @@ var plotParameters ={
...
+    var check2yvarplots = ['quiver', 'rose', 'ts_diagram'];
+    
     if(!this._validateTimeSpan(options)) return false;
-    if(options.plotType.toString().indexOf("_scatter")!=-1) return this._selectedX._validateNumberOfSelected("scatter",options.yvar[0][0])
-    else return this._selectedX._validateNumberOfSelected(options.plotType,options.yvar[0][0])
+    
+    if (options.plotType.toString().indexOf("_scatter")!=-1){
+      var var1 = $("#yvar1-select :selected").text();
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat could we use `this.$el.find('#yvar1-select :selected')` instead?

File: ooiui/static/js/views/common/SVGView.js [hunk 3]
```diff
@@ -241,13 +262,13 @@ var SVGView = Backbone.View.extend({
          // this should be uframe alert here.
          self.$el.html(' ');
          if (e.status == 400){
-         $('#plot-view').append('<div id="warning" class="alert alert-danger fade in"><a href="#" class="close" data-dismiss="alert">×</a><strong> Error:' + e.status +' </strong></div>');
+         $('#plot-view').append('<div id="warning" class="alert alert-danger fade in"><a href="#" class="close" data-dismiss="alert">×</a><strong> Error:' + e.status +' </strong>' +e.responseText+ '</div>');
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat could the use `this.$el.find(...)`?

File: ooiui/static/js/views/common/SVGView.js [hunk 4]
```diff
@@ -419,6 +460,7 @@ var SVGPlotView = SVGView.extend({
       this.useScatter = options.useScatter.toString();
       this.useEvent = options.useEvent.toString();
       this.plotType = options.plotType;
+      console.log(options.start_date)
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat remove console log pls

File: ooiui/static/js/views/common/SVGView.js [hunk 5]
```diff
@@ -408,6 +430,25 @@ var SVGPlotView = SVGView.extend({
 
         this.yvariable = not_list.join();
         this.dpa_flag = "1"     //this.getDpaFlag(options.xvar)
+      }else if (options.plotType == 'ts_diagram'){
+        this.yvariable = $( "#yvar2-select option:selected").val() + ',' + $( "#yvar1-select option:selected").val()
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat could we use `this.$el.find(..)` here?

File: ooiui/static/js/views/common/SVGView.js [hunk 6]
```diff
@@ -557,7 +601,14 @@ var SVGPlotControlView = Backbone.View.extend({
         }
 
   },
-
+  updateYVarDropdown: function(){
+      // This function copies parameters from parameters_id to the 3 yVar 
+      // dropdowns for the advanced plots
+      $( "#yvar1-select").html($("#parameters_id").html());
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat could we use `this.$el.find(...)` here?

File: ooiui/static/js/views/common/SVGView.js [hunk 7]
```diff
@@ -578,18 +629,88 @@ var SVGPlotControlView = Backbone.View.extend({
 
     if (plotType=="Time Series"){
       this.$el.find('#xVarTooltip').attr('data-original-title',"Time Series plot for selected parameter.  You may overlay up to 6 other parameters.")
+
+      $('#parameters_id').attr('disabled', false);
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat could we use `this.$el.find(...)` here?

**Comment by DanielJMaher** [comment]
@birdage How much do we want to leave in for you to clean when you get your hands in here?

File: ooiui/static/js/views/common/SVGView.js [hunk 8]
```diff
@@ -578,18 +629,88 @@ var SVGPlotControlView = Backbone.View.extend({
...
+      this.$el.find('#yvar2-selection').show();
+      this.$el.find('#yvar2-select-text').text("Y-Component");
+      this.$el.find('#yvar3-selection').show();
+      this.$el.find('#yvar3-select-text').text("Color");
     }
-
+  },
+  yVar1Change: function(e) {
+    var selectedParam = this.$el.find( "#yvar1-select option:selected").val()
+    $("#parameters_id").val(selectedParam);
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat could we use `this.$el.find(...)` here?

File: ooiui/static/js/views/common/SVGView.js [hunk 9]
```diff
@@ -601,7 +722,6 @@ var SVGPlotControlView = Backbone.View.extend({
     console.log(data.xvar);
```
Comment (birdage) [inline ooiui/static/js/views/common/SVGView.js]: @Bobfrat remove console log pls

File: ooiui/templates/science/plotting.html [hunk 10]
```diff
@@ -362,6 +363,7 @@
         options.xvar = 'time';
       }
       
+      console.log(options.yvar)
```
Comment (birdage) [inline ooiui/templates/science/plotting.html]: @Bobfrat remove console log pls

Comment (Bobfrat) [inline ooiui/templates/science/plotting.html]: Ok i get it, remove ```console.logs``` and replace ```this.$el.find(...)```.

Whats the advantage of using ```this.$el.find(...)```?

Comment (birdage) [inline ooiui/templates/science/plotting.html]: only searches in the given context/view so you wont bleed to the whole document :+1:

**Comment by birdage** [comment]
@Bobfrat nj! Could we remove the large comment section here https://github.com/Bobfrat/ooi-ui/blob/plotting_updates/ooiui/static/js/views/common/SVGView.js#L140 as well pls :smile:

File: ooiui/templates/science/plotting.html (hunk 10 above)
Comment (Bobfrat) [inline ooiui/templates/science/plotting.html]: Nice! Is that a backbone feature?

Comment (Bobfrat) [inline ooiui/templates/science/plotting.html]: Also you want me to remove all the console.logs in those files or just the ones I added?

Comment (birdage) [inline ooiui/templates/science/plotting.html]: @Bobfrat yeah, cool right!, yes to remove them all! i didnt see any more! haha

**Comment by Bobfrat** [comment]
@birdage I couldnt update all of the ```this.$el.find(``` instances you pointed out, but i did where I could.

**Comment by birdage** [comment]
@Bobfrat thats perfect, i guessed you wouldn't be able to do them all

**Comment by Bobfrat** [comment]
@birdage this one should be ready

**Comment by birdage** [comment]
:+1: will merge in a sec

**Pull request merged**